wp-includes/components/class-wp-service-worker-configuration-component.php

```diff
@@ -78,8 +78,21 @@ public function get_script() {
 
 		$script .= sprintf( "workbox.core.setCacheNameDetails( %s );\n", wp_service_worker_json_encode( $cache_name_details ) );
 
-		// @todo Add filter controlling workbox.skipWaiting().
-		// @todo Add filter controlling workbox.clientsClaim().
+		/**
+		 * Filters whether the service worker should update automatically when a new version is available.
+		 *
+		 * For optioning out from skipping waiting and displaying a notification to update instead, you could do:
+		 *
+		 *     add_filter( 'wp_service_worker_skip_waiting', '__return_false' );
+		 *
+		 * @param bool $skip_waiting Whether to skip waiting for the Service Worker and update when an update is available.
+		 */
+		$skip_waiting = apply_filters( 'wp_service_worker_skip_waiting', true );
+		if ( true === $skip_waiting ) {
+			$script .= "workbox.skipWaiting();\n";
+			$script .= "workbox.clientsClaim();\n";
```

Comment:
In [this guide](https://developers.google.com/web/tools/workbox/guides/generate-complete-sw#skip_waiting_and_clients_claim) updating seems to use these together, thought that this would be the standard way since it doesn't mention the cases for using `skipWaiting` separately, although it is possible.

According to Jeff Posnick's [answer here](https://stackoverflow.com/questions/51715127/what-are-the-downsides-to-using-skipwaiting-and-clientsclaim-with-workbox):
> `clientsClaim` ensures that all uncontrolled clients (i.e. pages) that are within scope will be controlled by a service worker immediately after that service worker activates. If you don't enable it, then they won't be controlled until the next navigation.

Also, on [this documentation page](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle#clientsclaim) it mentions: 
> If you use your service worker to load pages differently than they'd load via the network, clients.claim() can be troublesome, as your service worker ends up controlling some clients that loaded without it.

and also
> Note: I see a lot of people including clients.claim() as boilerplate, but I rarely do so myself. It only really matters on the very first load, and due to progressive enhancement the page is usually working happily without service worker anyway.

I guess it is better to have the option to filter those separately!

Reply:
Updated within 8cbc81f.